kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/scala/Option.java

```diff
@@ -0,0 +1,87 @@
+/**
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.streamnative.pulsar.handlers.kop.scala;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+import java.util.function.Supplier;
+
+public class Option<T> {
```

Comment:
This class seems a subset of `java.util.Optional`. What's the differentce?

Reply:
The only difference is the `match` method that is only used in `TransactionCoordinator#handleInitProducerId`. The initial purpose is to use `Option#match` method everywhere. But I found it could lead to many code changes later.

I think for now, we can change it back to Java's `Optional`.